nomad/structs/secure_variables.go

```diff
@@ -228,9 +228,13 @@ func NewRootKey(algorithm EncryptionAlgorithm) (*RootKey, error) {
 	switch algorithm {
 	case EncryptionAlgorithmAES256GCM:
 		key := make([]byte, 32)
```

Comment:
Maybe make the byte length shared between the length and the check. 1/2
```suggestion
		const keyBytes = 32
		key := make([]byte, keyBytes)
```